### PR TITLE
[10.0] [FIX] account: make the computation of taxes included in price but not affecting base correct

### PR DESCRIPTION
**Pull request by kebeclibre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by gustavovalverde** [comment]
Just to have it related, this is a Fix for: https://github.com/odoo/odoo/issues/10538

As the author @elicoidal, I'm keeping you in the loop.

**Comment by gustavovalverde** [comment]
@kebeclibre and @qdp-odoo, this PR https://github.com/odoo/odoo/pull/16790 also seems related, maybe also check that one.

File: addons/account/models/account.py [hunk 1]
```diff
@@ -821,6 +834,31 @@ def _fix_tax_included_price(self, price, prod_taxes, line_taxes):
...
+            return currency.round(tax_amount)
+
+    def _filter_taxes_included_base_not_affected(self, list_taxes, base, round_tax, currency, precision):
+        # taxes = dict_taxes.iltered(lambda t: t.price_include and not t.include_base_amount and t.amount_type == 'percent')
+        curr_taxes = self.filtered(lambda t: t.price_include and not t.include_base_amount and t.amount_type == 'percent').sorted(key=lambda r: r.sequence)
+        if not curr_taxes:
+            return list_taxes, None
+
+        taxes_cumul_rate = sum(curr_taxes.mapped(lambda t: t.amount))
+        excluded_amount = base / (1 + (taxes_cumul_rate / 100))
```
Comment (elicoidal) [inline addons/account/models/account.py]: Technically if taxes_cumul_rate is -100 you have a div by zero popping up

File: addons/account/models/account.py [hunk 2]
```diff
@@ -821,6 +834,31 @@ def _fix_tax_included_price(self, price, prod_taxes, line_taxes):
...
 
+    def _rounding_preference(self, tax_amount, bool_round_tax, currency, precision):
+        if not bool_round_tax:
+            return round(tax_amount, precision)
+        else:
+            return currency.round(tax_amount)
+
+    def _filter_taxes_included_base_not_affected(self, list_taxes, base, round_tax, currency, precision):
+        # taxes = dict_taxes.iltered(lambda t: t.price_include and not t.include_base_amount and t.amount_type == 'percent')
+        curr_taxes = self.filtered(lambda t: t.price_include and not t.include_base_amount and t.amount_type == 'percent').sorted(key=lambda r: r.sequence)
```
Comment (nim-odoo) [inline addons/account/models/account.py]: I would filter before, and call the method with the taxes to recompute

**Comment by gustavovalverde** [comment]
Hi everyone, any updates here?

**Comment by edsersolis** [comment]
Hello, when is this going to be fixed? Most restaurants work with multiple taxes included in the price, so this is a big issue.

**Comment by gustavovalverde** [comment]
@qdp-odoo I'd like to implement this before on my repo, is it safe?

**Comment by rajeeshknambiar** [comment]
This issue is a blocker for many of our customers now that India has the new ‘GST’ system. It needs the tax amount to be split into central & state taxes. This PR helps a lot, but falls short a bit in relation with issue #17974 - here an additional tax must be added on top of gst (which is a group).
This PR allows setting a group tax to affect the base for subsequent taxes, but code at line 779 `base = ret['base'] if tax.include_base_amount else base` still sets the base to tax excluded value.
Would be great to fix this shortcoming too.

**Comment by qdp-odoo** [comment]
Hello guys,

just to let you know that a PR fixing this issue (https://github.com/odoo/odoo/pull/18275) is about to be merged in v9. 
It would be great if you could test it and give us a feedback whether it is solving your issue (without breaking any of your use cases) before.

Thanks

**Comment by qdp-odoo** [comment]
the issue has been fixed by 
 the merge commit https://github.com/odoo/odoo/commit/5afb5ecd04cf49be6bedb655b187b19a65e1e02d of the superseeding PR #18275

**Pull request closed without merging**